test/file_input/typescript.d.ts

```diff
@@ -31,3 +31,54 @@ interface PaginationResult<T> {
   items: Array<T>
   total_items: number
 }
+
+/** Test integer */
+export const CONST_TEST_1 = 0;
+
+/** Shouldn't compile but should convert */
+export const CONST_TEST_2 = 0.0;
+
+/** Valid Rust but not valid typescript would be misleading if it made it into normal string ? */
+export const CONST_TEST_3 = b"Hello";
+
+/** Test serde_json */
+export const SERDE_JSON_1 = { "a" : "b" };
+
+export const SERDE_JSON_2 = { "a" : "b" };
+
+/**
+ * Variants should to discriminated unions
+ * The last serde/attribute combo matching the tag should be taken
+ */
+type Message =
+  /** Per Enum case Docs One */
+  | {
+      last_precedent: "unit-case-left",
+    }
+  /** Per Enum case Docs Two */
+  | {
+      last_precedent: "request-long-take",
+      id: string
+      method: string
+      params: number
+    }
+  | {
+      last_precedent: "response",
+      id: string
+      result: Date
+    };
+
+/**
+ * Integer enums should follow rust discrimination if literals (doesn't evaluate expression)
+ * The case renaming defaults to SCREAMING_SNAKE_CASE
+ */
+enum Foo {
+  BAR = 0,
+  BAZ = 123,
+  QUUX = 124,
+}
```

Comment:
I ran a short test with `serde` and `serde_json`:

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9b8515f62818783b8f8f0984426aecf3)
```
use serde::{Serialize, Deserialize};
use serde_json;

#[derive(Debug, Serialize, Deserialize)]
enum Operations {
    Add,
    Subtract,
}

#[derive(Debug, Serialize, Deserialize)]
enum Operations2 {
    Add = 10,
    Subtract = 100,
}

fn main() {
    let json = serde_json::to_string(&Operations::Add).unwrap();
    println!("{}", json);
    // ==> "Add"

    let json = serde_json::to_string(&Operations2::Subtract).unwrap();
    println!("{}", json);
    // ==> "Subtract"

}
```

It seems that regardless of whether enums are manually assigned numerical values or not doesn't matter -- it always returns a string. There is an [alternative serializer/deserializer](https://serde.rs/enum-number.html) that uses numbers, but the default just uses strings. What do you think about just using string values for the generated typescript enum?

Reply:
yes you're right and actually got shown this video recently [here](https://www.youtube.com/watch?v=0fTdCSH_QEU) where it says Typescript enums are probably not that helpful mainly because they move away from JS. I think the string values makes sense and I would be happy with:
`type Operations = "add" | "subtract"`